src/scripts/import-needs-assessment-data/add-regions.ts

```diff
@@ -0,0 +1,241 @@
+      if (isFulfilled(result)) {
+        return result.value;
+      } else {
+        return result.reason;
+      }
+    })
+    .reduce((resultsMap, workflowResult) => {
+      if (workflowResult.status) {
+        resultsMap[workflowResult.status].push(workflowResult);
+      } else {
+        resultsMap[UploadWorkflowStatus.OTHER].push(workflowResult);
+      }
+      return resultsMap;
+    }, resultsMap);
+
+  console.log("Add Geo.Regions results:");
+  Object.keys(resultsMap).forEach((key) => {
+    console.log(`     ${key}: ${resultsMap[key].length}`);
+
+    // NOTE: uncomment & set the status key to debug different types of results
+    // if (key !== UploadWorkflowStatus.SUCCESS && key !== UploadWorkflowStatus.ALREADY_EXISTS) {
+    //   resultsMap[key].forEach((result) => {
+    //     console.log(result)
+    //     console.log("\n")
+    //   })
+    // }
+  });
+
+  console.log("Adding items completed!");
+
+  const validRegions: Region[] = [
+    ...resultsMap[UploadWorkflowStatus.SUCCESS],
+    ...resultsMap[UploadWorkflowStatus.ALREADY_EXISTS],
+  ].reduce((regions: Region[], workflow: RegionUploadWorkflow) => {
+    return [...regions, workflow.data];
+  }, [] as Region[]);
+
+  return validRegions;
+}
+
+const isFulfilled = <T>(
+  value: PromiseSettledResult<T>,
+): value is PromiseFulfilledResult<T> => {
+  return value.status === "fulfilled";
+};
+
+const _isRejected = <T>(
+  value: PromiseSettledResult<T>,
+): value is PromiseRejectedResult => {
+  return value.status === "rejected";
+};
+
+/*  Consolidate Regions
+ * ------------------------------------------------------- */
+function consolidateRegions(data: NeedAssessment[]): string[] {
+  const regions = data.reduce((acc: string[], item) => {
+    const region = item.place?.region;
+    if (region && !acc.includes(region)) {
+      acc.push(region);
+    }
+    return acc;
+  }, []);
+
+  return regions;
+}
+
+/*  Parse Region
+ * ------------------------------------------------------- */
+function parseRegion({
+  data,
+  orig,
+  status,
+  logs,
+}: RegionUploadWorkflow): RegionUploadWorkflow {
+  logs = [...logs, `Log: parsing region "${orig}"`];
+
+  if (orig == null || typeof orig !== "string") {
+    throw {
+      data,
+      orig,
+      status: UploadWorkflowStatus.ORIGINAL_DATA_INVALID,
+      logs: [
+        ...logs,
+        `Error: Invalid region input: "${orig}". Expected a non-null value.`,
+      ],
+    };
+  }
+
+  const newData = {
+    ...data,
+    region: orig,
+  };
+
+  return {
+    data: newData,
+    orig,
+    status,
+    logs,
+  };
+}
+
+/*  Get Region
+ * ------------------------------------------------------- */
+async function getRegion({
+  data,
+  orig,
+  status,
+  logs,
+}: RegionUploadWorkflow): Promise<RegionUploadWorkflow> {
+  logs = [...logs, `Log: Checking if Geo.Region "${orig}" already exists.`];
+
+  //Fetch the data from Strapi
+  const response = await fetch(`${STRAPI_ENV.URL}/regions?`, {
```

Comment:
Do we need the `?` at the end of the endpoint here for query parameters? Can we just use `/regions` to get all regions and then if we need a specific one we can do something like `/regions/{region.Name}`.

Unless we are thinking of having some additional filtering on this endpoint

Reply:
That makes sense to me but I'm not entirely sure if there's any filtering that may need to occur in the future. 
@jtfairbank , thoughts?

Reply:
@ll-zerr should be fine to take the `?` out for now since we're trying to get all regions.  If we do need to filter later on then we can add it back in since we'd need to modify the code to support the filter query anyways.  I doubt we'll need to for regions tho, as there are just a handful of them and we add support for new regions slowly.  For a dataset like our Products filtering would make more sense, or perhaps making `get` requests to check individual items before upload instead of requesting the whole dataset to check locally.